cloudinit/config/cloud-init-schema.json

```diff
@@ -543,6 +543,244 @@
+                "default": false,
+                "description": "Set ``true`` to upgrade packages. Happens before install. Default: ``false``"
+            },
+            "package_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to reboot the system if required. Default: ``false``"
+            },
+            "apt_update": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_update``. Default: ``false``"
+            },
+            "apt_upgrade": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_upgrade``. Default: ``false``"
+            },
+            "apt_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_reboot_if_required``. Default: ``false``"
+            }
+        }
+    },
+    "cc_phone_home": {
+      "type": "object",
+      "properties": {
+        "phone_home": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["url"],
+          "properties": {
+            "url": {
+              "type": "string",
+              "description": "The URL to send the phone home data to."
+            },
+            "post": {
+              "description": "A list of keys to post or ``all``. Default: ``all``",
+              "oneOf": [
+                {"enum": ["all"]},
+                {
+                  "type": "array",
+                  "items": {
+                    "type": "string",
+                    "enum": [
+                      "pub_key_dsa",
+                      "pub_key_rsa",
+                      "pub_key_ecdsa",
+                      "pub_key_ed25519",
+                      "instance_id",
+                      "hostname",
+                      "fqdn"
+                    ]
+                  }
+                }
+              ]
+            },
+            "tries": {
+              "type": "integer",
+              "description": "The number of times to try sending the phone home data. Default: ``10``",
+              "default": 10
+            }
+          }
+        }
+      }
+    },
+    "cc_power_state_change": {
+      "type": "object",
+      "properties": {
+        "power_state": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["mode"],
+          "properties": {
+            "delay": {
+              "description": "Time in minutes to delay after cloud-init has finished. Can be ``now`` or an integer specifying the number of minutes to delay. Default: ``now``",
+              "default": "now",
+              "oneOf": [
+                {"type": "integer"},
+                {"type": "string", "pattern": "^\\+*[0-9]+$"},
```

Comment:
What is the purpose of this regex and string type? Do we have documented usage of this as a string anywhere?

Reply:
I want to "quietly" deprecate this syntax by just removing reference to it in the docs. If you look at [current docs](https://cloudinit.readthedocs.io/en/latest/topics/modules.html#power-state-change), it's a +number format because that's traditional `shutdown` command syntax. We can still support the syntax, but to me it makes more sense to just provide an integer. 

Good point that it should still be documented somewhere, but json files don't support comments, so not sure the best place to put that.

Reply:
> I want to "quietly" deprecate this syntax by just removing reference to it in the docs. If you look at [current docs](https://cloudinit.readthedocs.io/en/latest/topics/modules.html#power-state-change), it's a +number format because that's traditional `shutdown` command syntax. We can still support the syntax, but to me it makes more sense to just provide an integer.

Makes sense, thanks. I think the regex might be incorrect: `\\` means an escaped `\`, but I think you want to escape the `+`, correct? Also we shouldn't need the `*` since multiple plus signs (`++2`) isn't a valid notation.  Based on that, I think that `^\+[0-9]+$` is probably what we want. I haven't tested this with jsonschema specifically, but I have locally.

Reply:
I'd agree that we probably want to deprecate that "+\d" value from schema at some point to create concise/simple scheme for expected values. If we want to deprecate it fully, we can start logging warnings for uses of the "+\d" value in another PR when we handle proper deprecation notices.  Maybe we can add a checklist item for deprecation warnings for "delay" to SC-774

Reply:
> Since your tests pass, I guess that the \\ should be fine - this appears consistent with re.search()'s behavior and is a side effect of python str handling (though it wouldn't match as a PCRE, I don't think).

Looks like backslash is an escape character in json (e.g., you can "\n" in json), so the double backslashes should be correct. Agree with the rest of the regex, though I think `?` should work instead of `{0,1}`, correct?

Reply:
> Looks like backslash is an escape character in json (e.g., you can "\n" in json), so the double backslashes should be correct.

Today I learned, thanks for digging into this :)

> Agree with the rest of the regex, though I think `?` should work instead of `{0,1}`, correct?

Agreed